docs/learn/advanced/17-core.md

```diff
@@ -0,0 +1,134 @@
+
+## Header Service
+
+The header service provides access to the current block header.  This is useful for modules that
+need to access the block header fields like `Time` and `Height` during transaction processing.
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/a3729c1ad6ba2fb46f879ec7ea67c3afc02e9859/core/header/service.go#L11-L23
+```
+
+### Custom Header Service
+
+Core's service oriented architecture (SOA) allows for chain developers to define a custom
+implementation of the `HeaderService` interface.  This would involve creating a new struct that
+satisfies `HeaderService` but composes additional logic on top.  An example of where this would
+happen (when using depinject is shown below).  Note this example is taken from `runtime/v2` but
+could easily be adapted to `runtime/v1` (the default runtime 0.52).  This same pattern can be
+replicated for any core service.
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/489aaae40234f1015a7bbcfa9384a89dc8de8153/runtime/v2/module.go#L262-L288
+```
+
+These bindings are applied to the `depinject` container in simapp/v2 as shown below.
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/489aaae40234f1015a7bbcfa9384a89dc8de8153/simapp/v2/app_di.go#L72-L74
+```
+
+## Query and Message Router Service
+
+Both the query and message router services are implementation of the same interface, `router.Service`.
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/core/v1.0.0-alpha.4/core/router/service.go#L11-L16
+```
+
+Both are exposed to modules so that arbitrary messages and queries can be routed to the
+appropriate handler.  This powerful abstraction allows module developers to fully decouple
+modules from each other by using only the proto message for dispatching.   This is particularly
+useful for modules like `x/accounts` which require a dynamic dispatch mechanism in order to
+function.
+
+## TransactionService
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/core/v1.0.0-alpha.4/core/transaction/service.go#L21-L25
+```
+
+The transaction service provides access to the execution mode a state machine transaction is
+running in, which may be one of `Check`, `Recheck`, `Simulate` or `Finalize`.  The SDK primarily
+uses these flags in ante handlers to skip certain checks while in `Check` or `Simulate` modes,
+but module developers may find uses for them as well.
+
+## KVStore Service
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/core/v1.0.0-alpha.4/core/store/service.go#L5-L11
+```
+
+The KVStore service abstracts access to, and creation of, key-value stores.  Most use cases will
+be backed by a merkle-tree store, but developers can provide their own implementations if
+needed.  In the case of the `KVStoreService` implementation provided in `Environment`, module
+developers should understand that calling `OpenKVStore` will return a store already scoped to
+the module's prefix.  The wiring for this scoping is specified in `runtime`.
```

Comment:
I couldn't find any docs for `runtime` to link out to, do we want some?

Reply:
Yes, we need an explanation of what is runtime in concepts too 👍🏾 
I think the order should be, baseapp, transaction, core, runtime and then the rest, but we can do the ordering later.